compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java

```diff
@@ -6407,6 +6407,11 @@ private void rewriteInvocation(BLangInvocation invocation, boolean async) {
         reorderArguments(invocation);
 
         rewriteExprs(invocation.requiredArgs);
+        if (!invocation.requiredArgs.isEmpty() && invocation.langLibInvocation) {
```

Comment:
Can a lang lib invocation requiredArgs be empty at desugar?

Reply:
`string concat = a.concat("s","w");` if we take this case, 
https://github.com/ballerina-platform/ballerina-lang/blob/eac8c96aa024c308022b55c059e78532f19b30b9/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java#L6937
`parameterCountForPositionalArgs` is zero because it takes rest parameters. 
https://github.com/ballerina-platform/ballerina-lang/blob/eac8c96aa024c308022b55c059e78532f19b30b9/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java#L6962 
So, according to the above logic `requiredArgs` is set to zero but the `argExpr` is available.

Updated 432bb143f32cb73fd9de02b7a096c99feeca73f1